### DISCUSSION TITLE
关于使用范式的最佳实践

### DISCUSSION THREAD

**Discussion by secsilm** [milvus-io/milvus#31759] [Q&A and General discussion] [answered]
我今天把你们文档基本上都读了一下，发现在创建、插入、搜索时大概有两种写法：

1. 创建 client 然后使用 client 进行创建、插入和搜索等操作，即 `client.create_collection`、`client.insert`、`client.search`等。文档中绝大部分都是使用的这种方式。
2. 使用 connections连接然后使用 Collection 等对象进行插入和搜索等操作。在 example applications 中基本是使用的这种方式。

我想知道这两者有什么区别吗？在多进程/多线程的环境中，更推荐使用哪种？有没有什么最佳实践之类的？

谢谢！

**Comment by yhmo** [comment]
唯一的区别：
Milvus 1.x版本使用的是 client=MilvusClient(),  client.create_collection()、client.insert()、client.search()
Milvus 2.x版本使用的是 connections.connect(), collection = Collection(), colllection.insert(), collection.search()

**Comment by yhmo** [comment]
请各位提问时尽量使用英语，Milvus社区里有世界各地的工程师，用英文问答所有人都能看得懂。

**Comment by xiaofan-luan** [comment, marked as answer]
> 我今天把你们文档基本上都读了一下，发现在创建、插入、搜索时大概有两种写法：
> 
> 1. 创建 client 然后使用 client 进行创建、插入和搜索等操作，即 `client.create_collection`、`client.insert`、`client.search`等。文档中绝大部分都是使用的这种方式。
> 2. 使用 connections连接然后使用 Collection 等对象进行插入和搜索等操作。在 example applications 中基本是使用的这种方式。
> 
> 我想知道这两者有什么区别吗？在多进程/多线程的环境中，更推荐使用哪种？有没有什么最佳实践之类的？
> 
> 谢谢！

请参考2.4的官方文档，MilvusClient是我们现在推荐的用法。
ORM，也就是Collection.Insert()这一套，会一直维护直到3.0
